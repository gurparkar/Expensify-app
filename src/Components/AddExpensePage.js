import React from 'react'
import ExpenseForm from './ExpenseForm'

import { connect } from 'react-redux'
import { startAddExpense} from '../actions/expenses'

export class AddExpensePage extends React.Component {
      onSubmit =  (expense) => {
             
        this.props.onSubmit(expense)
        this.props.history.push('/')
    }

    render () {
            return (
                <div>
                <p> Add Expense</p>
                 <ExpenseForm 
                   onSubmit = {this.onSubmit}
                 />
             </div>
            )

    }
     
}


const mapDispatchToProps = (dispatch) => {
      return {
          onSubmit: (expense) => dispatch(startAddExpense(expense))
      }
}

export default connect(undefined, mapDispatchToProps )(AddExpensePage)

// const AddExpensePage = (props) => (
//     <div>
//        <p> Add Expense</p>
//         <ExpenseForm 
//          onSubmit = { (expense) => {
//              console.log(expense)
//              props.dispatch(addExpense(expense))
//              props.history.push('/')
//          }}
//         />
//     </div>
// )



