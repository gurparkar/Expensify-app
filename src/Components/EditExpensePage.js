import React from 'react'
import { connect } from 'react-redux'

import ExpenseForm from './ExpenseForm'
import { editExpense, removeExpense } from '../actions/expenses'

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense)
    this.props.history.push('/')
    console.log("Updated :", expense)
    
  }
  onRemove = () => {
    this.props.removeExpense({id: this.props.expense.id})
    this.props.history.push('/')
}

 render () {
     return (
        <div>
          <ExpenseForm
             expense = {this.props.expense}
             onSubmit = { this.onSubmit  }
          />
          <button onClick = { this.onRemove  }> Remove </button>
        </div>
           )
        } 
}

const mapStateToProps = (state,props) => {
      return {
          expense: state.expenses.find((expense) => expense.id === props.match.params.id)
      }  
}

const mapDispatchToProps = (dispatch) => {
      return {
            editExpense: (id, expense) => dispatch(editExpense(id, expense)),
            removeExpense: (data) => dispatch(removeExpense(data))
      }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)


// const EditExpensePage = (props) => {
//     console.log("Props here is : ",props)
//     return (
//         <div>
//               <ExpenseForm
//                  expense = {props.expense}
//                  onSubmit = {(expense) => {
//                       props.dispatch(editExpense(props.expense.id, expense))
//                       props.history.push('/')
//                       console.log("Updated :", expense)
                      
//                     }}
//               />

//               <button onClick = {}> Remove </button>
//         </div>
//     )
// }