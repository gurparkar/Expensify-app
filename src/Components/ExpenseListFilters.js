import React from 'react'
import { connect } from 'react-redux'
import 'react-dates/initialize'
import 'react-dates/lib/css/_datepicker.css';

import { DateRangePicker } from 'react-dates'
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters'

export class ExpenseListFilters extends React.Component {
      
      state = {
            focusedInput: null
      }

      onTextChange = (e) => {
                  this.props.setTextFilter(e.target.value)

      }

      onSortChange = (e) => {
            if(e.target.value == 'date') {
                  this.props.sortByDate()

            } else if (e.target.value == 'amount'){
                this.props.sortByAmount()
            }
      }

      onDatesChange = ({startDate, endDate}) => {
            this.props.setStartDate(startDate)
            this.props.setEndDate(endDate)
      }



      render (){
            return (
                  <div>
                   <input type = "text"   value ={this.props.filters.text} onChange = {this.onTextChange } />
                   <select value = {this.props.filters.sortBy} onChange = { this.onSortChange }>
                          <option value = 'date'>Date</option>
                          <option value = 'amount'>Amount</option>
                   </select>
                   
                   <DateRangePicker  
                     startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
                     startDateId="your_unique_start_date_id" // PropTypes.string.isRequired,
                     endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
                     endDateId="your_unique_end_date_id" // PropTypes.string.isRequired,
                     onDatesChange={this.onDatesChange} // PropTypes.func.isRequired,
                     showClearDates = {true}
                    focusedInput={this.state.focusedInput} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                    onFocusChange={focusedInput => this.setState({ focusedInput })}
                    numberOfMonths = {1}
                    isOutsideRange = {() => false }
              / >
           
                   </div>
               )
      }

}



const mapStateToProps = (state) => {
      return {
            filters: state.filters
      }
}

const mapDispatchToProps = (dispatch) => {
      return {
             setTextFilter : (text) => dispatch(setTextFilter(text)),
             sortByDate : () => dispatch(sortByDate()),
             sortByAmount : () => dispatch(sortByAmount()),
             setStartDate : (startDate) => dispatch(setStartDate(startDate)),
             setEndDate: (endDate) => dispatch(setEndDate(endDate))

      }
}

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)