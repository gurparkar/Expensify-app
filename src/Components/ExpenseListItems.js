import React from 'react'
import moment from 'moment'
//import { connect } from 'react-redux'

import { Link } from 'react-router-dom'
import numeral from 'numeral'
export const ExpenseListItems = ({ id,description, amount, createdAt}) => {
    
    return(
    <div>
    <Link to = {`/edit/${id}`}>
        <h3>{description}</h3>
        
    </Link>
    
    <p>{ numeral(amount).format('$0,0.00') }
     - 
     { moment(createdAt).format('MMMM Do YYYY') }</p>


    </div>
    )
}

// const mapStateToProps = (state) => {
//       return {
//              expenses: state.expenses
//       }
// }



export default ExpenseListItems