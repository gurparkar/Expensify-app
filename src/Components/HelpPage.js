import React from 'react'

const HelpPage = () => (
    <div>
        This is from my dashboard component Help page .. . . .  . ..
    </div>
)

export default HelpPage