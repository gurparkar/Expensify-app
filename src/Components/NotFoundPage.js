import React from 'react'

const NotFoundPage = () => (
    <div>
        404 Page - <Link to="/">Go Home</Link>
    </div>
)

export default NotFoundPage