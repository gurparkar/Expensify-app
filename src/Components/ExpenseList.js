import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItems from './ExpenseListItems'
import SelectExpenses from '../selectors/expenses'


export const ExpenseList = (props) => {
     return (
         <div>
             { props.expenses.length == 0 ? 
                <p>No expenses</p>: 
                props.expenses.map((expense) => {
                    return <ExpenseListItems key = {expense.id} {...expense} />
              })

            }

         </div>
     )
}

const mapStateToProps = (state) => {
    return {
         expenses: SelectExpenses(state.expenses, state.filters),
 
    }
} 

const ConnectedExpenseList = connect(mapStateToProps)(ExpenseList)

export default ConnectedExpenseList