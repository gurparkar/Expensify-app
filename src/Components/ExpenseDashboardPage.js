import React from 'react'
import  ExpenseList from './ExpenseList'

import ExpenseListFilters from './ExpenseListFilters'

const ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        This is from my dashboard component
        <ExpenseList />
    </div>
)

export default ExpenseDashboardPage