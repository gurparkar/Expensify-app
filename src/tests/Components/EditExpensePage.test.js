import React from 'react'
import { shallow } from 'enzyme'

import { EditExpensePage } from '../../Components/editExpensePage'
import Expenses from '../../../expected'



let editExpense, history, wrapper,removeExpense





beforeEach( () => {
    editExpense = jest.fn()
    removeExpense = jest.fn()
    history = { push: jest.fn() }
    wrapper = shallow( <EditExpensePage editExpense= {editExpense} history = {history}  removeExpense = {removeExpense } expense = {Expenses.expenses[2]}/>)
})

test ( 'Render EditExpensePage react Component correctly in homepage', () => {
     expect(wrapper).toMatchSnapshot()  
})

test ( 'Render EditExpensePage react Component correctly', () => {
     wrapper.find('ExpenseForm').prop('onSubmit')(Expenses.expenses[2])
     expect(history.push).toHaveBeenLastCalledWith('/')
     expect(editExpense).toHaveBeenLastCalledWith(Expenses.expenses[2].id, Expenses.expenses[2])
})

test ( 'Render removeExpense react Component correctly in homepage', () => {
     wrapper.find('button').simulate('click')
     expect(history.push).toHaveBeenLastCalledWith('/')
     expect(removeExpense).toHaveBeenLastCalledWith({
           id: Expenses.expenses[2].id
     })
})



// beforeEach( () => {
//     editExpense = jest.fn()
//     history = { push: jest.fn() }
//     removeExpense = jest.fn()
//     wrapper = shallow( < EditExpensePage 
//         editExpense = {editExpense} 
//         removeExpense = { removeExpense}
//         history = {history}  />)
// })

// test ( 'Render EditExpense Component correctly in homepage', () => {
//      expect(wrapper).toMatchSnapshot()  
// })