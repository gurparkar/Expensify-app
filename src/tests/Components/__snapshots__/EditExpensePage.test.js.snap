// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render EditExpensePage react Component correctly in homepage 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 4500,
        "createdAt": 345600000,
        "description": "Credit Card",
        "id": "3",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
     Remove 
  </button>
</div>
`;
